src/main/scala/io/iohk/ethereum/domain/package.scala

```diff
@@ -2,20 +2,29 @@ package io.iohk.ethereum
 
 import akka.util.ByteString
 import io.iohk.ethereum.mpt.{ByteArrayEncoder, ByteArraySerializable, HashByteArraySerializable, MerklePatriciaTrie, NodesKeyValueStorage}
-import io.iohk.ethereum.rlp.UInt256RLPImplicits._
+import io.iohk.ethereum.rlp.RLPImplicits._
+import io.iohk.ethereum.utils.ByteUtils
+import org.spongycastle.util.BigIntegers
 
 package object domain {
 
-  val byteArrayUInt256Serializer = new ByteArrayEncoder[UInt256] {
-    override def toBytes(input: UInt256): Array[Byte] = input.bytes.toArray[Byte]
+  val byteArrayBigIntSerializer = new ByteArrayEncoder[BigInt] {
+    override def toBytes(input: BigInt): Array[Byte] =
+      ByteUtils.padLeft(ByteString(BigIntegers.asUnsignedByteArray(input.bigInteger)), 32).toArray[Byte]
```

Comment:
I don't think this will work correctly:
```
scala> BigIntegers.fromUnsignedByteArray(BigIntegers.asUnsignedByteArray(BigInt(-1).bigInteger))
res3: java.math.BigInteger = 255
```

Also, why padding?

Reply:
and for that conversion - I believe you only need to convert keys one way, to byte array

Reply:
> padding is needed for Ethereum support

What exactly do you mean?

Reply:
Ethereum expects serialized keys to be 32-bytes long (in previous implementation see: `input.bytes.toArray[Byte]` - `UInt256.bytes` adds the padding

Reply:
Wait, exactly 32 bytes or at least?

Reply:
For EVM it's exactly 32-bytes (bounded by usage of uint256). For other VM's it can be shorter or longer, BUT, if we want to keep a single implementation, it's the only place we can add this padding for EVM support